src/coreclr/jit/emitxarch.h

```diff
@@ -202,6 +202,16 @@ bool IsWEvexOpcodeExtension(const instrDesc* id)
 
     instruction ins = id->idIns();
 
+    if (IsRexW1Instruction(ins))
+    {
+        return true;
+    }
+    else if (IsRexWIGInstruction(ins) || IsRexW0Instruction(ins))
+    {
+        // TODO: Make this a simple assert once all instructions are annotated
+        return false;
+    }
+
```

Comment:
I plan on covering this `TODO` in a follow-up PR (notably will be the "next" thing I do).

Reply:
Do we have any systematic way to improve upon the `INS_has_wbit` flag and the opcode extension bit flag?

I agree that we can move the opcode extension bit into the instruction table, but the `TakesRexWBit` check is still pretty complicated and requires a lot of contextual information.

If we eventually move everything to checks on `instrDesc` instead of `ins`, we probably can simplify this quite a bit. That is later down the road though.

Reply:
I can take a look as part of this. It would indeed be nicer/simpler to be able to handle this "everywhere".

I imagine many of the GPR instructions need something like `REX_WX` bit that indicates its 0 or 1 based on the EA_ATTR size or alternatively some way to split the instructions between `D` and `Q` versions.

Reply:
Have a prototype of tracking the REX.W bit stuff for SIMD via https://github.com/dotnet/runtime/pull/83473